theseus/geometry/se3.py

```diff
@@ -284,26 +284,88 @@ def vee(matrix: torch.Tensor) -> torch.Tensor:
         SE3._hat_matrix_check(matrix)
         return torch.cat((matrix[:, :3, 3], SO3.vee(matrix[:, :3, :3])), dim=1)
 
-    def _transform_shape_check(self, point: Union[Point3, torch.Tensor]):
-        raise NotImplementedError
-
     def _copy_impl(self, new_name: Optional[str] = None) -> "SE3":
         return SE3(data=self.data.clone(), name=new_name)
 
     # only added to avoid casting downstream
     def copy(self, new_name: Optional[str] = None) -> "SE3":
         return cast(SE3, super().copy(new_name=new_name))
 
+    def _transform_shape_check(self, point: Union[Point3, torch.Tensor]):
+        err_msg = "SE3 can only transform 3-D vectors."
+        if isinstance(point, torch.Tensor):
+            if not point.ndim == 2 or point.shape[1] != 3:
+                raise ValueError(err_msg)
```

Comment:
The error message here might be ambiguous if the user passes a single point without a batch dim.

Reply:
Let's change the message for 

```python
err_msg = f"SE3 can only transform vectors of shape [batch_size, 3], but input has shape {point.shape}."
```

Reply:
Fixed.